### PR TITLE
File listing

### PR DESCRIPTION
**Pull request by red-tt-bear**
Moved routes into routes so they use base path
Added development feature to look at directory listings
Added package param to prune or not to prune
Made development start docker not prune the dependencies

### CODE REVIEW

File: src/configurations.ts [hunk 1]
```diff
@@ -111,6 +111,7 @@ const configurations = {
     app: {
         nodeEnv: nodeEnv,
         isProduction: isProduction,
+        isDevelopment: nodeEnv === 'development',
```
Comment (PanopticaRising) [inline src/configurations.ts:114]: Shouldn't this be !isProduction?

File: src/puppetmaster.ts [hunk 2]
```diff
@@ -93,7 +94,7 @@ export default class PuppetMaster {
 
         logger.debug(`Navigating to ${urlPath}.html`);
         // The Express server statically hosts the tmp files.
-        await page.goto(`http://127.0.0.1:${configurations.server.port}/export/${urlPath}`, {waitUntil: ['load', 'networkidle0'], timeout: configurations.puppeteer.navigationTimeout});
+        await page.goto(urljoin(`http://127.0.0.1:${configurations.server.port}/`, configurations.server.basePath, `/tmp/${urlPath}`), {waitUntil: ['load', 'networkidle0'], timeout: configurations.puppeteer.navigationTimeout});
```
Comment (PanopticaRising) [inline src/puppetmaster.ts:97]: is `/tmp` always prepended? I thought that was just our default `tempDirectory`

Comment (PanopticaRising) [reply src/puppetmaster.ts:97]: Oh, I see below, the `use('/tmp`

**Review by PanopticaRising — APPROVED** [review]

File: src/configurations.ts (hunk 1 above)
Comment (red-tt-bear) [reply src/configurations.ts:114]: I did that first but then changed my mind, figured it made sense that this was explicitly a development feature so for those that don't specify they don't get this feature or prod features

File: src/puppetmaster.ts (hunk 2 above)
Comment (red-tt-bear) [reply src/puppetmaster.ts:97]: Yeah, before this was set up on /export but the default base path was /export so it was a little confusing
I thought /tmp made more sense to explain what it is actually doing

**Pull request merged**